CLN: remove unneeded try/except in nanops

topper-123 opened the pull request:
Look like this try/except isn't needed, checking in CI if it verifies the result.

topper-123 committed:
CLN: remove unneded try/except in nanops

jbrockmendel commented on pandas/core/nanops.py:
```diff
@@ -1075,11 +1075,8 @@ def reduction(
-            except (AttributeError, TypeError, ValueError):
-                result = np.nan
+            result = getattr(values, meth)(axis, dtype=dtype_max)
+            result.fill(np.nan)
```
do we have tests that get here?

topper-123 committed:
clean up min/max empty cases

topper-123 committed:
remove return type

topper-123 commented:
I rearranged it again. `if (axis is not None and values.shape[axis] == 0) or values.size == 0:` is always the same as `if values.size == 0:` AFAIKT.

So I've just added a `if values.size == 0` clause at the top and cleaned up even further, so it's quite simple now.

mroeschke commented on pandas/core/nanops.py:
```diff
@@ -1067,22 +1067,14 @@ def reduction(
         axis: AxisInt | None = None,
         skipna: bool = True,
         mask: npt.NDArray[np.bool_] | None = None,
-    ) -> Dtype:
```
Can we keep this type annotation?

topper-123 replied:
This annotation was wrong and this doesn't return a `Dtype` but the the result of a min/max of a DataFrame/Series, i.e. typically a scalar or a ndarray (but the min/max of a Series can in principle be anything....)

IDK if we return type `Scalar` as annotation for Series?

mroeschke replied:
Yeah not either, but since this is an inner function the typing here probably doesn't matter as much

mroeschke commented on pandas/core/nanops.py:
```diff
@@ -1067,22 +1067,14 @@ def reduction(
-        dtype = values.dtype
+    ):
+        if values.size == 0:
+            return _na_for_min_count(values, axis)
```
I assume we have tests that get here?

topper-123 replied:
The `values.size == 0` case is actually never reached in the current code:

You can see this function is wrapped in `bottleneck_switch` and if you go to `bottleneck_switch` you can see (line 125) it takes care of the `values.size == 0` case, also when `use_bottleneck = False` (`kwds.get("min_count") is None` is always true for `_nanminmax`).

I just think this is more readable than the old code.

mroeschke replied:
Gotcha. Thanks for the explanation

mroeschke commented:
Thanks @topper-123
